### PR TITLE
Promote `ManagedIstio` and `APIServerSNI` feature gates

### LINKED ISSUE #2942: Graduate APIServerSNI (+ ManagedIstio) feature gate
**How to categorize this issue?**

/area control-plane
/area cost
/area scalability
/kind enhancement
/priority normal

**What would you like to be added**:
The `APIServerSNI` and `ManagedIstio` (usually used together) should be graduated to **beta** after we have collected sufficient experience and test results on regularly tested and busy landscapes. Any potential issues and blockers graduating this feature can be referenced or documented here.

Steps:
- [x] Enable feature gates on development landscape 
- [x] Enable feature gates on landscape featuring extended automated tests
- [x] Enable feature gates on landscape with stakeholder workload (large landscape)
- [x] Enable Prometheus metrics and Grafana Dashboards for `ManagedIstio`
- [x] Add `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` environment variables to all Gardener components to avoid going to through the `apiserver-proxy`
- [x] Replace `nginx` with `envoy` to have the same proxy on both the client and server side. Enovy supports [upstream proxy protocol transport](https://www.envoyproxy.io/docs/envoy/v1.16.0/intro/arch_overview/other_features/ip_transparency.html?highlight=envoy%20filters%20listener%20proxy_protocol#proxy-protocol)
- [x] Add Prometheus scraping for `apiserver-proxy`
- [x] Expose `apiserver-proxy`'s `envoy` metrics` available for scraping.
- [x] Allow for end-users to specify the `istio-ingress-gateway` namespace and name.
- [x] Add `PodDisruptionBudget` for `istio-ingress-gateway` and `istiod`
- [x] Add a `MutatingWebhookConfiguration` for `Pods` in the Shoot cluster, adding `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT`. It'll run as a sidecar to the API Server (to not have any extra latency to all `CREATE` requests) with `FailurePolicy: Ignore` and very low `timeoutSeconds` - (1 or 2)
- [x] Set KEEPALIVE on both `apiserver-proxy` and `istio-ingressgateway`
- [x] ~Change the port of `kube-apiserver` gateways to `9443` to prevent issues if `istio-ingressgateway` service has port `443` is removed.~
- [x] ~Preserve original source IP~ GA feature

/cc @mvladev

**Why is this needed**:
Both feature gates are an essential milestone to host control planes in a cost efficient and scalable manner.

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area open-source
/kind enhancement
/priority normal

**What this PR does / why we need it**:
The `ManagedIstio` and `APIServerSNI` feature gates in the `gardenlet` have been promoted to beta.
Along the way, I updated some feature gate owners.

**Which issue(s) this PR fixes**:
Fixes #2942

**Special notes for your reviewer**:
/squash

**Release note**:

```action operator
The `ManagedIstio` and `APIServerSNI` feature gates in the `gardenlet` have been promoted to beta and are now enabled by default. If you run your own istio installation then you have to disable the `ManagedIstio` feature gate (and probably also the `APIServerSNI`) in your gardenlet configurations.
```

### CODE REVIEW

File: pkg/gardenlet/features/features.go [hunk 1]
```diff
@@ -28,9 +28,9 @@ var (
 		features.Logging:                {Default: false, PreRelease: featuregate.Alpha},
 		features.HVPA:                   {Default: false, PreRelease: featuregate.Alpha},
 		features.HVPAForShootedSeed:     {Default: false, PreRelease: featuregate.Alpha},
-		features.ManagedIstio:           {Default: false, PreRelease: featuregate.Alpha},
+		features.ManagedIstio:           {Default: false, PreRelease: featuregate.Beta},
 		features.KonnectivityTunnel:     {Default: false, PreRelease: featuregate.Alpha},
-		features.APIServerSNI:           {Default: false, PreRelease: featuregate.Alpha},
+		features.APIServerSNI:           {Default: false, PreRelease: featuregate.Beta},
```
Comment (rfranzke) [inline pkg/gardenlet/features/features.go]: I did not enable them by default yet. Should we do it?

Comment (vpnachev) [reply pkg/gardenlet/features/features.go]: Yes.

Comment (vpnachev) [reply pkg/gardenlet/features/features.go]: Also, enable them by default here https://github.com/gardener/gardener/blob/5eefe2db2e64db1d7947c3bdf605ac7e7879346b/example/20-componentconfig-gardenlet.yaml#L88-L90

**Review by vpnachev — CHANGES_REQUESTED** [review]
Please, update also https://github.com/gardener/gardener/blob/master/docs/deployment/feature_gates.md#feature-gates-for-alpha-or-beta-features.

**Comment by mvladev** [comment]
Hold on - end-users might already have Istio installed on their clusters, enabling it by default might  cause problems for them.

**Comment by vpnachev** [comment]
What is the plan then? I am thinking about deprecating the `ManagedIstio` and moving it to an extension, wdyt?

**Comment by rfranzke** [comment]
I agree with @mvladev. My gud feeling is that it's not a good idea to enable it by default yet. 

> What is the plan then? I am thinking about deprecating the ManagedIstio and moving it to an extension, wdyt?

😕 Why? At least at this point, what is the benefit? I vote for being pragmatic now: Promote the feature gates to beta to indicate that we consider it as "mature enough to be used", but do not enable it by default to not interfere with existing installations.

**Comment by mvladev** [comment]
> I agree with @mvladev. My gud feeling is that it's not a good idea to enable it by default yet.
> 
> > What is the plan then? I am thinking about deprecating the ManagedIstio and moving it to an extension, wdyt?
> 
> 😕 Why? At least at this point, what is the benefit? I vote for being pragmatic now: Promote the feature gates to beta to indicate that we consider it as "mature enough to be used", but do not enable it by default to not interfere with existing installations.

I would agree with @vpnachev . I would rather have the isito installation of of the gardener. Now that I think about it - probably would do the same for the `GardenerSeedScheduler` feature gate as well. Or we keep it beta, but do not enable it.

**Comment by rfranzke** [comment]
Yes, generally, I agree as well, but I don't see how this is related now and why we would need to hold the beta promotion now. Actually, our (or at least mine) goal with this PR is only to indicate that the feature is mature enough and "battle-tested" so that others can feel comfortable enough to try it out as well. IMO this does not rule out the opportunity to move everything out of g/g in the future when we have the time and necessity to do so?
OTOH, I am also totally fine with keeping everything as it is (no beta promotion) because we anyways never followed this process, so it probably doesn't matter too much to consumers of Gardener (I assume, many are using "alpha" gates like `Logging`, `HVPA`, and so on for a long time already).
What do you prefer?

**Comment by vpnachev** [comment]
I would prefer just to try to establish some policy like k8s. Alpha feature are disabled by default and needs to be explicitly enabled. Beta and GA features are enabled by default and needs to be explicitly disabled.

Thinking again about https://github.com/gardener/gardener/pull/3633#issuecomment-787428033, I actually do not see it as a blocker. We just need to add an `action required` release note and ask the users that already have Istio in their seed clusters to explicitly disable the `ManagedIstio` feature gate.

**Comment by rfranzke** [comment]
OK, I've now enabled them by default and added a corresponding release note. Please check.

**Review by timebertt — DISMISSED** [review]
/lgtm
I think, this is the way to go.

If we want to allow users to disable the gardener managed istio installation after graduating the feature gate to GA, we can still add a configuration option, which is then more fitting for having optional components than a feature gate in my opinion...

**Review by vpnachev — DISMISSED** [review]
/lgtm

We could enable them by default here https://github.com/gardener/gardener/blob/ef495eb662556cdb7dff4401eaa32ddc539e604b/example/20-componentconfig-gardenlet.yaml#L88-L90 but it can also be done with some other PR.

**Comment by rfranzke** [comment]
@vpnachev done

**Review by timebertt — DISMISSED** [review]
/lgtm

**Review by vpnachev — DISMISSED** [review]
Thanks!
/lgtm

**Comment by rfranzke** [comment]
The `release-v1.18` branch is already created, so this PR won't make it into the v1.18 release. I've pushed 31656b1 which changes the docs to `v1.19`.

**Review by vpnachev — APPROVED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**